### PR TITLE
Custom Capabilities

### PR DESCRIPTION
**Pull request by ashby-bowles**
This PR is intended to sync the implementation of custom capabilities with conductor mobile using an `onCapabilitiesCreated` method.

### CODE REVIEW

**Comment by konwinkler** [comment]
I see the PR is for a merge into master. Right now we are using git flow to stream line the conductor development. For this the process is to make a PR for develop first and then later for a release make a PR from a develop release branch to master.

File: src/main/java/io/ddavison/conductor/Locomotive.java [hunk 1]
```diff
@@ -66,6 +79,34 @@ public void init() {
         }
     }
 
+
+    protected DesiredCapabilities onCapabilitiesCreated(DesiredCapabilities desiredCapabilities) {
+        return desiredCapabilities;
+    }
+
+    private DesiredCapabilities getCapabilities(ConductorConfig configuration) {
```
Comment (konwinkler) [inline src/main/java/io/ddavison/conductor/Locomotive.java]: This private method does not seem to add any functionality. Could the calls to this be replaced by `buildCapabilities` and this method removed?

Comment (ashby-bowles) [reply src/main/java/io/ddavison/conductor/Locomotive.java]: Oh yes, removed 👍

File: src/main/java/io/ddavison/conductor/util/DriverUtil.java [hunk 2]
```diff
@@ -62,7 +67,7 @@ public static WebDriver getDriver(ConductorConfig config) {
                     break;
                 case EDGE:
                     EdgeOptions edgeOptions = new EdgeOptions();
-                    setCustomCapabilities(config, edgeOptions);
+                    setCustomCapabilities(config, edgeOptions, desiredCapabilities);
```
Comment (jossjacobo) [inline src/main/java/io/ddavison/conductor/util/DriverUtil.java]: You are missing the `capabilities = `, combine this and the next line, like we did on the `SAFARI` case.

File: src/main/java/io/ddavison/conductor/util/DriverUtil.java [hunk 3]
```diff
@@ -98,12 +103,22 @@ public static WebDriver getDriver(ConductorConfig config) {
         return driver;
     }
 
-    public static void setCustomCapabilities(ConductorConfig config, MutableCapabilities capabilities) {
+    public static MutableCapabilities setCustomCapabilities(ConductorConfig config, MutableCapabilities capabilities, DesiredCapabilities customDesiredCapabilities) {
```
Comment (jossjacobo) [inline src/main/java/io/ddavison/conductor/util/DriverUtil.java]: We are missing yet _*another*_ look for `MutableCapabilities capabilities`, I think this is way the test is failing. Before this method would just adding the `config` custom caps into the `capabilities` object. Now we are creating a new Capabilities object and we need to add the caps from the config, the original caps, and the new desired caps.

**Comment by jossjacobo** [comment]
@ashby-bowles the tests seem to be stuck on teamcity.

**Pull request merged**